lib/src/layout/content_area.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:macos_ui/src/layout/scaffold.dart';
+
+import '../../macos_ui.dart';
+
+/// The widget that fills the rest of the body of the macOS [Scaffold].
+///
+/// A [Scaffold] can contain only one [ContentArea].
+class ContentArea extends StatelessWidget {
+  /// Creates a widget that fills the body of the scaffold.
+  /// The [builder] can be null to show an empty widget.
+  ///
+  /// The width of this
+  /// widget is automatically calculated in [ScaffoldScope].
+  const ContentArea({Key? key, @required this.builder}) : super(key: key);
+
+  /// The builder that creates a child to display in this widget, which will
+  /// use the provided [_scrollController] to enable the scrollbar to work.
+  ///
+  /// Pass the [_scrollController] obtained from this method, to a scrollable
+  /// widget used in this method to work with the internal [Scrollbar].
+  final ScrollableWidgetBuilder? builder;
+
+  static final _scrollController = ScrollController();
```

Comment:
I see here you're considering an application to have only one scaffold, when there can be more. I think the `Scrollbar` should be implemented by the dev, not by the `Scaffold` widget

Reply:
Also, you mention in the documentation of the property `builder` that the dev can use [_scrollController], but the dev can't because the property is private

Reply:
By default every section of the macOS window is scrollable. I've also heard that flutter considers adding scrollbar to every scrollview by default. I'll track that and update our repo later.
Check here for the scrollController example: https://github.com/GroovinChip/macos_ui/blob/4ce9eb81779acd8e9d971cbe5f189148c5409f8e/example/lib/main.dart#L53

And about having one Scaffold, that's what I've observed so far from the official mac apps. It's up to the developer to customize the UI though. The ScaffoldScope takes the nearest Scaffold, so I think we can have multiple Scaffold.
I'll add the sidebar buttons together with the tabbed navigation after this PR lands. Consider this the basic frame.
Scaffold is a long work in progress. We can't wait to factor in everything before merging this. It'll take forever